pkg/db/db.go

```diff
@@ -67,11 +62,15 @@ func (db *DB) Save(key string, val []byte, seq uint64) {
 	if seq == seqDeleted {
 		panic("reserved seq")
 	}
-	if rec, ok := db.Records[key]; ok && seq == rec.Seq && bytes.Equal(val, rec.Val) {
+	// If data is discarded, we assume key identifies data (data hash).
```

Comment:
How frequently are we calling `db.Save()` with the unchanged value? If not too frequently, maybe it's better to err on the safe side and return early **only** if `db.dataDiscarded == false`?